lib/evmone/instructions.hpp

```diff
@@ -173,13 +173,33 @@ inline void signextend(ExecutionState& state) noexcept
     const auto ext = state.stack.pop();
     auto& x = state.stack.top();
 
-    if (ext < 31)
+    if (ext < 31)  // For 31 we also don't need to do anything.
     {
-        auto sign_bit = static_cast<int>(ext) * 8 + 7;
-        auto sign_mask = uint256{1} << sign_bit;
-        auto value_mask = sign_mask - 1;
-        auto is_neg = (x & sign_mask) != 0;
-        x = is_neg ? x | ~value_mask : x & value_mask;
+        const auto e = ext[0];  // uint256 -> uint64.
+        const auto sign_word_index =
+            static_cast<size_t>(e / sizeof(e));      // Index of the word with the sign bit.
+        const auto sign_byte_index = e % sizeof(e);  // Index of the sign byte in the sign word.
+        const auto sign_word = x[sign_word_index];
```

Comment:
also random thought, no idea how compilers would react:
```suggestion
        auto& sign_word = x[sign_word_index];
```
then remove `sext_word` variable and directly assign
```
sign_word = sext | value;
```

Reply:
I applied this change, thanks. The output assembly is the same just the order of instructions changes a bit.